frontend/src/RecordingPageComp/PlaybackContainerComp.jsx

```diff
@@ -1,69 +1,273 @@
-
   setNote,
   setPlaybackDuration,
 }) {
+  const [midiNotes, setMidiNotes] = useState([]);
+  const [isPlaying, setIsPlaying] = useState(false);
+
   const audioContextRef = useRef(new AudioContext());
 
   function frequencyToMIDINoteNumber(frequency) {
-    setNote(Math.round(69 + 12 * Math.log2(frequency / 440)));
+    return Math.round(69 + 12 * Math.log2(frequency / 440));
   }
 
   useEffect(() => {
     handlePitchDetection(selectedBlob);
+    stopPlayback();
   }, [selectedBlob]);
 
   const handlePitchDetection = (blob) => {
     if (!audioContextRef.current) {
       audioContextRef.current = new AudioContext();
     }
-
-    console;
-
     if (!blob) {
       return;
     } else {
-      // Use blob.arrayBuffer() method to read blob as ArrayBuffer
+      const sampleRate = audioContextRef.current.sampleRate;
       blob
         .arrayBuffer()
         .then((arrayBuffer) => {
-          audioContextRef.current.decodeAudioData(arrayBuffer, (buffer) => {
-            const input = buffer.getChannelData(0); // mono audio
-            const detector = PitchDetector.forFloat32Array(input.length);
-            const [pitch] = detector.findPitch(
-              input,
-              audioContextRef.current.sampleRate
-            );
-            frequencyToMIDINoteNumber(pitch);
-            setPlaybackDuration(buffer.duration);
-          });
+          audioContextRef.current.decodeAudioData(
+            arrayBuffer,
+            (buffer) => {
+              let input = buffer.getChannelData(0); // mono audio
+
+              // Pre-processing: Simple normalization
+
+              let maxVal = 0;
+              for (let i = 0; i < input.length; i++) {
+                const absValue = Math.abs(input[i]);
+                if (absValue > maxVal) {
+                  maxVal = absValue;
+                }
+              }
+              if (maxVal > 0) {
+                input = input.map((x) => x / maxVal);
+              }
+
+              //Multipitch Detector
+              const windowSize = 1024;
+              const hopSize = 512;
+              const pitches = analyzePitchOverTime(
+                input,
+                sampleRate,
+                windowSize,
+                hopSize
+              );
+
+              const groupedPitches = groupPitches(pitches, sampleRate, hopSize);
+              const filteredPitches = filterShortDurations(groupedPitches, 0.1); // Filter out short durations
+              setMidiNotes(filteredPitches);
+
+              //Single Pitch detector
+              const singlePitchDetector = PitchDetector.forFloat32Array(
+                input.length
+              );
+              const [pitch] = singlePitchDetector.findPitch(
+                input,
+                audioContextRef.current.sampleRate
+              );
+              //Values will be passed on to SimilarSounds
+              setNote(frequencyToMIDINoteNumber(pitch));
+              setPlaybackDuration(buffer.duration);
```

Comment:
I would consider encapsulating the logic for single pitch detection and multi-pitch detection into two separate functions to increase code modularity